Source/DafnyLanguageServer/Workspace/Notifications/VerificationDiagnosticsParams.cs

```diff
@@ -0,0 +1,671 @@
+        }
+      }
+
+      // Fill in the missing "Unknown" based on the surrounding content
+      // The filling only takes Verified an Error
+      var previousNotUnknown = LineVerificationStatus.Unknown;
+      var lineDelta = 1;
+      // Two passes so that we can fill gaps based on what happened before AND after
+      for (var line = 0; 0 <= line; line += lineDelta) {
+        if (line == numberOfLines) {
+          lineDelta = -1;
+          previousNotUnknown = LineVerificationStatus.Unknown;
+          continue;
+        }
+        if (previousNotUnknown != LineVerificationStatus.Verified &&
+            previousNotUnknown != LineVerificationStatus.VerifiedObsolete &&
+            previousNotUnknown != LineVerificationStatus.VerifiedVerifying) {
+          previousNotUnknown = LineVerificationStatus.Unknown;
+        }
+        if (result[line] == LineVerificationStatus.Unknown) {
+          result[line] = previousNotUnknown;
+        } else {
+          previousNotUnknown = result[line];
+        }
+      }
+
+      var resolutionErrorRendered = 0;
+      foreach (var diagnostic in diagnostics) {
+        if (resolutionErrorRendered >= numberOfResolutionErrors) {
+          break;
+        }
+        result[diagnostic.Range.Start.Line] = LineVerificationStatus.ResolutionError;
+        resolutionErrorRendered++;
+      }
+
+      var existsErrorRange = false;
+      var existsError = false;
+      foreach (var line in result) {
+        existsErrorRange = existsErrorRange || line == LineVerificationStatus.ErrorRange;
+        existsError = existsError || line == LineVerificationStatus.Error;
+      }
+
+      if (existsErrorRange && !existsError) {
+        existsError = false;
+      }
+
+      return result;
+    }
+  }
+
+
+  public enum VerificationStatus {
+    Unknown = 0,
+    Verified = 200,
+    Inconclusive = 270,
+    Error = 400
+  }
+
+  public enum CurrentStatus {
+    Current = 0,
+    Obsolete = 1,
+    Verifying = 2
+  }
+
+  public enum LineVerificationStatus {
+    // Default value for every line, before the renderer figures it out.
+    Unknown = 0,
+    // For first-time computation not actively computing but soon. Synonym of "obsolete"
+    // (scheduledComputation)
+    Scheduled = 1,
+    // For first-time computations, actively computing
+    Verifying = 2,
+    VerifiedObsolete = 201,
+    VerifiedVerifying = 202,
+    // Also applicable for empty spaces if they are not surrounded by errors.
+    Verified = 200,
+    // For containers of other diagnostics nodes (e.g. methods)
+    ErrorRangeObsolete = 301,
+    ErrorRangeVerifying = 302,
+    ErrorRange = 300,
+    // For individual assertions in error ranges
+    ErrorRangeAssertionVerifiedObsolete = 351,
+    ErrorRangeAssertionVerifiedVerifying = 352,
+    ErrorRangeAssertionVerified = 350,
+    // For specific lines which have errors on it. They take over verified assertions
+    ErrorObsolete = 401,
+    ErrorVerifying = 402,
+    Error = 400,
```

Comment:
The names of the `ErrorRange...`, `ErrorRangeAssertion...`, and `Error...` constants don't seem super clear to me. I don't have a concrete suggestion for what they should be, though. :)

Reply:
Alternatives, please comment on them:
* I could add underscores to split names? For example `Error_Obsolete`.
* I could explicit what `Error` means: `Error_Current` or even `ErrorAssertionFailedCurrent`, because every error is an assertion failure. Or just `AssertionFailedCurruent`
* I could replace `ErrorRange` with `RangeWithError`. They indicate we are in the context of a method that has errors.
* I could replace `ErrorRange` with `ErrorContext` as well
* I could replace `ErrorRangeAssertionverified` with `AssertionFailedContextButAssertionsVerified`

As I developed the naming myself, I'm pretty biased and would love to hear your external feedback.

Reply:
Given those description, I now think I understand what they mean! :)

I think I like `AssertionFailedCurrent` (or even just `AssertionFailed`) and `ErrorContext`. Given those, maybe the last could be `AssertionVerifiedInErrorContext`?

Reply:
Nice ! I like all three.